src/modules/dashboard/components/Dialogs/GnosisControlSafeDialog/GnosisControlSafeDialog.tsx

```diff
@@ -63,25 +91,146 @@ type Props = DialogProps &
   Partial<WizardDialogType<object>> &
   ActionDialogProps;
 
-const validationSchema = (isPreview) =>
-  yup.object().shape({
+const GnosisControlSafeDialog = ({
+  colony,
+  cancel,
+  callStep,
+  prevStep,
+  isVotingExtensionEnabled,
+}: Props) => {
+  const [showPreview, setShowPreview] = useState(false);
+  const [selectedContractMethods, setSelectedContractMethods] = useState<{
+    [key: number]: AbiItemExtended | undefined;
+  }>();
+  const [expandedValidationSchema, setExpandedValidationSchema] = useState<
+    Record<string, any>
+  >({});
+
+  const getMethodInputValidation = useCallback(
+    (inputType: string, contractName: string, isArraySchema?: boolean) => {
+      if (inputType.slice(-2) === '[]') {
+        return yup.array().when('contractFunction', {
+          is: (contractFunction) => contractFunction === contractName,
+          then: yup
+            .array()
+            .ensure()
+            .of(
+              getMethodInputValidation(
+                inputType.slice(0, -2),
+                contractName,
+                true,
+              ),
+            ),
+          otherwise: false,
+        });
+      }
+      if (inputType === 'uint256' || inputType === 'int256') {
+        return yup.number().when('contractFunction', {
+          is: (contractFunction) =>
+            contractFunction === contractName || isArraySchema,
+          then: yup
+            .number()
+            .test(
+              'is-integer',
+              () => MSG.notIntegerError,
+              (value) => new Decimal(value || 0).isInteger(),
+            )
+            .test(
+              'is-integer-safe',
+              () => MSG.notSafeIntegerError,
+              (value) =>
+                new Decimal(value || 0).lte(new Decimal(MaxUint256.toString())),
+            )
```

Comment:
When I enter `115792089237316195423570985008687907853269984665640564039457584007913129639935`, which is the max safe integer, I get an error that it's not safe, so maybe the Decimal library doesn't handle these large numbers so well?

Reply:
We use Decimal for more complex calculations than this, so this ain't it. 

The value that arrives at this line of the code it's already not the same amount you inputted, from what I've tested it's coming rounded up, for some reason. So something is happening to the number earlier in the process.

Reply:
@ArmandoGraterol Will this be addressed in a separate issue?